### PR TITLE
John conroy/error pages

### LINKED ISSUE #1044: implement new error page design
Approved UI design for error pages is here: https://github.com/hubmapconsortium/portal-ui/issues/339

### PR DESCRIPTION
**Pull request by john-conroy**
Fix #1044

I'm going to handle the extra margin between the header and error page in a future PR before merging to master. Background color for maintenance pages is handled in #1082 .

Added in a vscode snippet because I kept copy and pasting the test imports and thought it could be useful for others to use as well.

Screen Shot 2020-08-19 at 4 23 27 PM

### CODE REVIEW

File: context/app/main.py [hunk 1]
```diff
@@ -5,12 +5,12 @@
 
 def bad_request(e):
     '''A 400 means the request to the API failed.'''
-    return render_template('errors/400.html', types={}), 400
+    return render_template('pages/base_react.html', flask_data={'errorCode': 400}), 400
```
Comment (mccalluc) [inline context/app/main.py:8]: Perhaps introduce helper function?
```
def render_react_error(code):
    return render_template('pages/base_react.html', flask_data={'errorCode': code}), code
```

File: context/app/static/js/components/error/ErrorBody/ErrorBody.jsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      <>
+        You have not been added to the HuBMAP Group on Globus. Request access at <HelpEmailLink />.
+      </>
+    );
+  }
+
+  if (errorCode === 403) {
+    return (
+      <>
+        You may not have access to this resource or you need to <LoginLink />.
```
Comment (mccalluc) [inline context/app/static/js/components/error/ErrorBody/ErrorBody.jsx:43]: Can we not suggest logging in if they're already logged in?

File: context/app/static/js/components/error/ErrorBody/ErrorBody.jsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    return (
+      <>
+        If this page should exist, submit a bug report to <HelpEmailLink />.
+      </>
+    );
+  }
+
+  return (
+    <>
+      If this problem persists, submit a bug report to <HelpEmailLink />.
```
Comment (mccalluc) [inline context/app/static/js/components/error/ErrorBody/ErrorBody.jsx:58]: We shouldn't reach this, right? but if we do, let's include the errorCode. Could get rid of the "If this problem persists"?
```suggestion
      Unexpected error "{errorCode}": Please submit a bug report to <HelpEmailLink />.
```

File: context/app/static/js/components/error/ErrorBody/ErrorBody.spec.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  render(<ErrorBody errorCode={404} />);
+  expect(screen.getByText('If this page should exist,', { exact: false })).toBeInTheDocument();
+  expect(screen.getByRole('link', { name: help.name })).toHaveAttribute('href', help.url);
+});
+
+test('all other statuses', () => {
+  render(<ErrorBody errorCode={500} />);
+  expect(screen.getByText('If this problem persists,', { exact: false })).toBeInTheDocument();
+  expect(screen.getByRole('link', { name: help.name })).toHaveAttribute('href', help.url);
+});
```
Comment (mccalluc) [inline context/app/static/js/components/error/ErrorBody/ErrorBody.spec.js:48]: I'm not sure that I'll keep up with you, but I really respect the level of detail you put into the tests. Thanks!

File: context/app/static/js/index.jsx [hunk 5]
```diff
@@ -3,44 +3,9 @@ import ReactDOM from 'react-dom';
...
-      <Footer />
-    </Providers>,
-    document.getElementById('react-footer'),
-  );
-}
+ReactDOM.render(
+  // eslint-disable-next-line no-undef
+  <App flaskData={flaskData} />,
+  document.getElementById('react-content'),
+);
```
Comment (mccalluc) [inline context/app/static/js/index.jsx:11]: yay!

File: context/app/static/js/pages/Error/Error.jsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    ? 'Portal unavailable for scheduled maintenance.'
+    : `HTTP Error ${errorCode}: ${title}`;
+
+  return (
+    <Background>
+      <StyledPaper>
+        <StyledTypography variant="h1" $mb={2}>
+          {title}
+        </StyledTypography>
+        <StyledTypography variant="subtitle1" color="primary" $mb={1}>
```
Comment (mccalluc) [inline context/app/static/js/pages/Error/Error.jsx:31]: In a recent PR, I just added the purple color to the `subtitle1` theme. Maybe, check if my moving the color there was appropriate? And this one is intended to be a special case?

File: context/app/static/js/components/error/ErrorBody/ErrorBody.jsx (hunk 2 above)
Comment (john-conroy) [reply context/app/static/js/components/error/ErrorBody/ErrorBody.jsx:43]: @tsliaw ?

Comment (tsliaw) [reply context/app/static/js/components/error/ErrorBody/ErrorBody.jsx:43]: If that's not too much additional work, that should be better to not suggest logging in if they're already logged in.

Comment (tsliaw) [reply context/app/static/js/components/error/ErrorBody/ErrorBody.jsx:43]: Text:

**Comment by tsliaw** [comment]
Catch all:
Title: Unexpected Error
Subtitle: HTTP Error <error_code>
Body: If this problem persists, submit a bug report to help@hubmapconsortium.org.

**Pull request merged**